### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by nozpheratu** [yuki24/did_you_mean#85]
did_you_mean was running into a segmentation fault on my Mac, see the relevant stack trace & crash log attached:

```
/Users/cyle/.rvm/gems/ruby-2.3.0@global/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb:10: [BUG] Segmentation fault at 0x0000000000014d
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin16]
```

[ruby_2017-04-18-170328_tacticas-MacBook-Pro.txt](https://github.com/yuki24/did_you_mean/files/931221/ruby_2017-04-18-170328_tacticas-MacBook-Pro.txt)

I was trying to access an undefined variable in a transition callback using the [Statesman](https://github.com/gocardless/statesman) gem, e.g:

```ruby
after_transition(to: :purchased) do |order, transition|
    undefined_variable.foo
end
```

**Comment by yuki24** [comment]
Ruby 2.3.0 has a bug. Please upgrade to Ruby 2.3.4 and did_you_mean 1.0.3. These versions only include bug fixes and are compatible with Ruby 2.3.0 and did_you_mean 1.0.0, so you should be able to upgrade easily. Thanks for reporting!

**Comment by betesh** [comment]
Duplicates #67 #72 #74
